File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int checkBalanced(TreeNode *root)
    {
        if(root==nullptr)
        {
            return 0;
        }

        int leftTree = checkBalanced(root->left);
        if(leftTree == -1)
        {
            return -1;
        }

         int rightTree = checkBalanced(root->right);
        if(rightTree == -1)
        {
            return -1;
        }

        if(abs(leftTree - rightTree) > 1 )
        {
            return -1;
        } 
        return max(leftTree,rightTree)+1;
        
    }

    bool isBalanced(TreeNode* root) {
        return checkBalanced(root) != -1;
    }
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    int checkBalanced(TreeNode root)
    {
        if(root == null)
        {
            return 0;
        }

        int leftTree = checkBalanced(root.left);
        if(leftTree == -1)
        {
            return -1;
        }

        int rightTree = checkBalanced(root.right);
        if(rightTree == -1)
        {
            return -1;
        }

        if(Math.abs(leftTree - rightTree) > 1)
        {
            return -1;
        }
        return Math.max(leftTree,rightTree)+1;
    }
    public boolean isBalanced(TreeNode root) {
        return checkBalanced(root)!=-1;
    }
}